packages/off-chain/README.md

```diff
@@ -1,3 +1,55 @@
 # off-chain
 
-run example script "doSomething.ts with `node-ts doSomething.ts`
+## Links
+Docker Image: mihirsinhparmar/ie-bac
+
+Run jon on bacalhau
+```
+bacalhau docker run mihirsinhparmar/ie-bac
```

Comment:
This command doesn't work for me "Error while executing job".

Is that the same amd64 issue?

This is what I get if I use `bacalhau describe ...`:

```
APIVersion: V1beta1
Metadata:
  ClientID: f4d3ca22f82233396e43ff64383a6f8a80e2a53915d8ff05660292bdd2e111bf
  CreatedAt: "2023-01-02T22:05:06.834907625Z"
  ID: e7faeac5-199d-470f-a417-2bbc2f0cfb06
Spec:
  Deal:
    Concurrency: 1
  Docker:
    Image: mihirsinhparmar/ie-bac
  Engine: Docker
  ExecutionPlan:
    ShardsTotal: 1
  Language:
    JobContext: {}
  Publisher: Estuary
  Resources:
    GPU: ""
  Sharding:
    BatchSize: 1
    GlobPatternBasePath: /inputs
  Timeout: 1800
  Verifier: Noop
  Wasm: {}
  outputs:
  - Name: outputs
    StorageSource: IPFS
    path: /outputs
Status:
  JobState:
    Nodes:
      QmVAb7r2pKWCuyLpYWoZr9syhhFnTWeFaByHdb8PkkhLQG:
        Shards:
          "0":
            NodeId: QmVAb7r2pKWCuyLpYWoZr9syhhFnTWeFaByHdb8PkkhLQG
            PublishedResults: {}
            State: Cancelled
            VerificationResult: {}
      QmXaXu9N5GNetatsvwnTfQqNtSeKAD6uCmarbh3LMRYAcF:
        Shards:
          "0":
            NodeId: QmXaXu9N5GNetatsvwnTfQqNtSeKAD6uCmarbh3LMRYAcF
            PublishedResults: {}
            State: Cancelled
            VerificationResult: {}
      QmYgxZiySj3MRkwLSL4X2MF5F9f2PMhAE3LV49XkfNL1o3:
        Shards:
          "0":
            NodeId: QmYgxZiySj3MRkwLSL4X2MF5F9f2PMhAE3LV49XkfNL1o3
            PublishedResults: {}
            State: Error
            Status: Received ActionComputeError from node QmYgxZiySj3MRkwLSL4X2MF5F9f2PMhAE3LV49XkfNL1o3
              that has not bid on this shard
            VerificationResult: {}
      QmdZQ7ZbhnvWY1J12XYKGHApJ6aufKyLNSvf8jZBrBaAVL:
        Shards:
          "0":
            NodeId: QmdZQ7ZbhnvWY1J12XYKGHApJ6aufKyLNSvf8jZBrBaAVL
            PublishedResults: {}
            State: Cancelled
            VerificationResult: {}
  Requester:
    RequesterNodeID: QmdZQ7ZbhnvWY1J12XYKGHApJ6aufKyLNSvf8jZBrBaAVL
    RequesterPublicKey: CAASpgIwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDVRKPgCfY2fgfrkHkFjeWcqno+MDpmp8DgVaY672BqJl/dZFNU9lBg2P8Znh8OTtHPPBUBk566vU3KchjW7m3uK4OudXrYEfSfEPnCGmL6GuLiZjLf+eXGEez7qPaoYqo06gD8ROdD8VVse27E96LlrpD1xKshHhqQTxKoq1y6Rx4DpbkSt966BumovWJ70w+Nt9ZkPPydRCxVnyWS1khECFQxp5Ep3NbbKtxHNX5HeULzXN5q0EQO39UN6iBhiI34eZkH7PoAm3Vk5xns//FjTAvQw6wZUu8LwvZTaihs+upx2zZysq6CEBKoeNZqed9+Tf+qHow0P5pxmiu+or+DAgMBAAE=
```

Reply:
yeah, this is something that I have been facing multiple times. That is, it is not deterministic.

It's not an issue of `amd64`.

When I run it, it runs.

Here's the log:
```
0xmihir@Mihirs-MacBook-Pro off-chain % bacalhau docker run mihirsinhparmar/ie-bac   
Job successfully submitted. Job ID: ace03c38-1ed8-411b-9ffd-276dacf9c256
Checking job status... (Enter Ctrl+C to exit at any time, your job will continue running):

               Creating job for submission ... done ✅
               Finding node(s) for the job ... done ✅
                     Node accepted the job ... done ✅
           Job finished, verifying results ... done ✅
              Results accepted, publishing ... Job Results By Node:

To download the results, execute:
  bacalhau get ace03c38-1ed8-411b-9ffd-276dacf9c256

To get more details about the run, execute:
  bacalhau describe ace03c38-1ed8-411b-9ffd-276dacf9c256
```

Reply:
now the weird part is, I run the command
`bacalhau docker run mihirsinhparmar/ie-bac`

the first time, it fails, the second time it runs
```
0xmihir@Mihirs-MacBook-Pro off-chain % bacalhau docker run mihirsinhparmar/ie-bac
Job successfully submitted. Job ID: d7d7abc2-03e2-47ec-b249-14a5f4a0c083
Checking job status... (Enter Ctrl+C to exit at any time, your job will continue running):

               Creating job for submission ... done ✅
               Finding node(s) for the job ... done ✅
                     Node accepted the job ... done ✅
           Error while executing the job.

Job Results By Node:

To download the results, execute:
  bacalhau get d7d7abc2-03e2-47ec-b249-14a5f4a0c083

To get more details about the run, execute:
  bacalhau describe d7d7abc2-03e2-47ec-b249-14a5f4a0c083
0xmihir@Mihirs-MacBook-Pro off-chain % bacalhau docker run mihirsinhparmar/ie-bac
Job successfully submitted. Job ID: b5ce2ca0-7f2f-4c78-8abe-2ac1815bdb87
Checking job status... (Enter Ctrl+C to exit at any time, your job will continue running):

               Creating job for submission ... done ✅
               Finding node(s) for the job ... done ✅
                     Node accepted the job ... done ✅
           Job finished, verifying results ... done ✅
              Results accepted, publishing ... Job Results By Node:

To download the results, execute:
  bacalhau get b5ce2ca0-7f2f-4c78-8abe-2ac1815bdb87

To get more details about the run, execute:
  bacalhau describe b5ce2ca0-7f2f-4c78-8abe-2ac1815bdb87
```

Reply:
Weird, I don't get it to successfully execute the job 🤔 

I rebuilt the docker container locally (`docker build ..`; `docker tag ..`), pushed it to `fabianschu/ex1` and now I can execute it succesfully via `bacalhau docker run fabianschu/ex1`. I'm a bit puzzled tbh. Isn't that exactly the same problem that you were facing initially? Feels like an issue on the bacalhau side of things. 

I don't think it should stop you from moving on with the implementation but I'm wondering if we should ask about it in the grant slack channel. Wdyt?

Reply:
Weird, I don't get it to successfully execute the job 🤔

I rebuilt the docker container locally with the shell script and now I can execute it succesfully via bacalhau docker run fabianschu/ex3. I'm a bit puzzled tbh. Isn't that exactly the same problem that you were facing initially? Feels like an issue on the bacalhau side of things.

I don't think it should stop you from moving on with the implementation but I'm wondering if we should ask about it in the grant slack channel. Wdyt?

Reply:
I have raised the question on slack channel.